src/RGUnityBots/Assets/Tests/Runtime/RGActionManagerTests.cs

```diff
@@ -82,5 +91,401 @@ public void TestValueRanges()
                 Assert.IsTrue(Mathf.Approximately(((Vector2)disc[3].MaxValue).y, 1.0f));
             }
         }
+
+        private GameObject FindGameObject(string name)
+        {
+            foreach (var t in UnityEngine.Object.FindObjectsOfType<Transform>(true))
+            {
+                if (t.gameObject.name == name)
+                {
+                    return t.gameObject;
+                }
+            }
+            return null;
+        }
+
+        private void FindAndPerformAction(string pathSuffix, object param)
+        {
+            var actionInst = RGActionManager.GetValidActions().FirstOrDefault(actionInst =>
+                actionInst.BaseAction.Paths[0].Last() == pathSuffix);
+            Debug.Assert(actionInst != null, $"Action {pathSuffix} is missing");
+            foreach (var inp in actionInst.GetInputs(param))
+            {
+                inp.Perform();
+            }
+        }
+
+        [UnityTest]
+        public IEnumerator TestActionManager()
+        {
+            if (Keyboard.current == null)
+            {
+                InputSystem.AddDevice<Keyboard>();
+            }
+            
+            SceneManager.LoadSceneAsync("ActionManagerTestScene", LoadSceneMode.Single);
+            yield return RGTestUtils.WaitForScene("ActionManagerTestScene");
+            
+            GameObject eventSystem = GameObject.Find("EventSystem");
+            var eventSys = eventSystem.GetComponent<EventSystem>();
+            
+            // Test Input System Keyboard
+            RGActionManager.StartSession(eventSys);
+            GameObject inputSysKeyListener = FindGameObject("InputSysKeyListener");
+            inputSysKeyListener.SetActive(true);
+            try
+            {
+                FindAndPerformAction("Keyboard.current.anyKey", true);
+                yield return null;
+                yield return null;
+                LogAssert.Expect(LogType.Log, "Keyboard.current.anyKey.wasPressedThisFrame");
```

Comment:
Nice! Glad we can use these as examples as well for monobehaviours that would need to interact with this API